### PR TITLE
Add back headers metadata pass in kafka subscribe

### PR DESCRIPTION
**Pull request by DeepanshuA**
Signed-off-by: Deepanshu Agarwal <deepanshu.agarwal1984@gmail.com>

# Description

Commit https://github.com/dapr/components-contrib/commit/e1697ba1159f8ae8493b6fec9f33d42cbb751efc had merged `master` into `feature/pubsub-batching` branch and thus adding https://github.com/dapr/components-contrib/pull/2017 to feature/pubsub-batching branch.
But, commit https://github.com/dapr/components-contrib/commit/8500da577c0cbd14165d88b5ce8c8b290a28eaf4 from PR #2113   mistakenly removed change done under #2017 . 

https://github.com/dapr/components-contrib/issues/2212 was filed as headers were no longer getting subscribed due to this.
This PR corrects that.

Currently, Cert test for Kafka won't be able to verify this part due to https://github.com/dapr/components-contrib/pull/2017#issuecomment-1245119239.
I have created an issue https://github.com/dapr/dapr/issues/5519 to add relevant test to e2e for this testing.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2212 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by DeepanshuA** [comment]
I see kafka certification test failing - checking this one.

**Comment by DeepanshuA** [comment]
> I see kafka certification test failing - checking this one.

Ok, now it passes but binding kafka cert test fails - kafka cert tests seem to be flaky.

**Review by mukundansundar — DISMISSED** [review]
lgtm

File: internal/component/kafka/consumer.go [hunk 1]
```diff
@@ -188,7 +188,13 @@ func (consumer *consumer) doCallback(session sarama.ConsumerGroupSession, messag
 		Topic: message.Topic,
 		Data:  message.Value,
 	}
-
+	// This is true only when headers are set (Kafka > 0.11)
+	if message.Headers != nil && len(message.Headers) > 0 {
```
Comment (ItalyPaleAle) [inline internal/component/kafka/consumer.go:192]: Nit:

```suggestion
	if len(message.Headers) > 0 {
```

(if `message.Headers` is `nil`, then its length is 0)

**Review by ItalyPaleAle — APPROVED** [review]

**Review by berndverst — APPROVED** [review]

**Pull request merged**